extensions/ql-vscode/src/stories/variant-analysis/VariantAnalysisAnalyzedRepos.stories.tsx

```diff
@@ -125,24 +125,22 @@ Example.args = {
 };
 
 faker.seed(42);
-const uniqueStore = {};
+const uniqueNumbers = [...Array(1000).keys()].sort(() => Math.random() - 0.5);
 
 const manyScannedRepos = Array.from({ length: 1000 }, (_, i) => {
   const mockedScannedRepo = createMockScannedRepo();
+  const randomInt = uniqueNumbers.pop();
+  const uniqueId =
+    randomInt === undefined ? Math.random() * 8000 + 1001 : randomInt;
 
   return {
     ...mockedScannedRepo,
     analysisStatus: VariantAnalysisRepoStatus.Succeeded,
     resultCount: faker.number.int({ min: 0, max: 1000 }),
     repository: {
       ...mockedScannedRepo.repository,
-      // We need to ensure the ID is unique for React keys
-      id: faker.helpers.unique(faker.number.int, [], {
-        store: uniqueStore,
-      }),
-      fullName: `octodemo/${faker.helpers.unique(faker.word.sample, [], {
-        store: uniqueStore,
-      })}`,
+      id: uniqueId,
+      fullName: `octodemo/${uniqueId}`,
```

Comment:
Any idea why we're generating our own unique id? Can't faker do this for us? E.g. 
```
`octodemo/${faker.string.uuid}`
```

I'm guessing it doesn't have to be a number, since it's in a string.

Reply:
Creating a unique number is what they deprecated, there are some discussions and the gist of it is that they ask people to create their own packages for this functionality.

Reply:
But do we need a number in this case?

Reply:
No, but I suppose the string wouldn't be unique either. I don't know what the likelihood is for creating two similar strings randomly.  We could use `https://fakerjs.dev/api/string.html#alpha` but we would have to check for uniqueness.

Reply:
I think I should add the uniqueness comment from Koen back in?!

Reply:
What happens if the keys aren't unique? Is the failure mode that it throws an error, or it silently does wrong stuff?

That will help us answer the question of how careful do we need to be. If React safely detects this case and tells us, this seems fine. The manual solution the user can take is to reload the page and get a new set of random numbers, and this is so easy to do that maybe it's not worth complicating the code.

Overall I don't mind. I've also left a few different suggestions on the other comment thread.

Reply:
> No, but I suppose the string wouldn't be unique either. 

Ah I didn't realise uniqueness was completely removed. In that case, then perhaps we should use `nanoid` like we do elsewhere (e.g. [here](https://github.com/github/vscode-codeql/blob/2f66be3fccb1ed0077acf20d0e790585e907e65e/extensions/ql-vscode/src/run-queries-shared.ts#L506))?

Agree about leaving the uniqueness comment though.

Reply:
Ah nice, we already use a library for unique ids! That sounds like a good solution!

Reply:
`nanoid` is also fine by me. 👍🏼 

For the record, as far as I know this is also just a random number generator. There's no guarantee of uniqueness, but the randomness is enough that the change of conflict is negligible. But that's fine for this context.